src/channel/internal.ts

```diff
@@ -0,0 +1,349 @@
+  existingFsmId?: string
+  url: string
+  role: 'initiator' | 'responder'
+  initiatorId: EncodedData<'ak'>
+  responderId: EncodedData<'ak'>
+  pushAmount: number
+  initiatorAmount: BigNumber
+  responderAmount: BigNumber
+  channelReserve?: BigNumber | number
+  signedTx?: string
+  ttl?: number
+  host: string
+  port: number
+  lockPeriod: number
+  existingChannelId?: string
+  offChainTx?: string
+  reconnectTx?: string
+  timeoutIdle?: number
+  timeoutFundingCreate?: number
+  timeoutFundingSign?: number
+  timeoutFundingLock?: number
+  timeoutSign?: number
+  timeoutAccept?: number
+  timeoutInitialized?: number
+  timeoutAwaitingOpen?: number
+  statePassword?: string
+  debug: boolean
+  sign: SignTxWithTag
+  offchainTx?: string
+}
+
+export interface ChannelHandler extends Function {
+  enter?: Function
+}
+
+export interface ChannelState {
+  signedTx: any
+  resolve: Function
+  reject: Function
+  sign: SignTx
+  handler?: ChannelHandler
+  onOnChainTx?: Function
+  onOwnWithdrawLocked?: Function
+  onWithdrawLocked?: Function
+  onOwnDepositLocked?: Function
+  onDepositLocked?: Function
+  closeTx?: string
+}
+
+export interface ChannelFsm {
+  handler: ChannelHandler
+  state?: ChannelState | {
+    resolve: Function
+    reject: Function
+  }
+}
+
+export interface ChannelMessage {
+  id?: number
+  jsonrpc: string
+  method: string
+  params: any
+  payload?: any
+  data?: any
+  error?: ChannelMessageError
+}
+
+interface ChannelMessageError {
+  code: number
+  message: string
+  data: [{
+    message: string
+    code: number
+  }]
+  request: ChannelMessage
+}
+
+// Send ping message every 10 seconds
+const PING_TIMEOUT_MS = 10000
+// Close connection if pong message is not received within 5 seconds
+const PONG_TIMEOUT_MS = 5000
+
+export const options = new WeakMap<object, ChannelOptions>()
+export const status = new WeakMap<object, string>()
+export const state = new WeakMap<object, string>()
+const fsm = new WeakMap<object, ChannelFsm>()
+const websockets = new WeakMap<object, W3CWebSocket>()
+export const eventEmitters = new WeakMap<object, EventEmitter>()
+const messageQueue = new WeakMap<object, string[]>()
+const messageQueueLocked = new WeakMap<object, boolean>()
+const actionQueue = new WeakMap<object, ChannelAction[]>()
+const actionQueueLocked = new WeakMap<object, boolean>()
+const sequence = new WeakMap<object, number>()
+export const channelId = new WeakMap<object, string>()
+const rpcCallbacks = new WeakMap<object, Map<number, Function>>()
+const pingTimeoutId = new WeakMap<object, NodeJS.Timeout>()
+const pongTimeoutId = new WeakMap<object, NodeJS.Timeout>()
+export const fsmId = new WeakMap<object, string>()
+
+export function emit (channel: {}, ...args: any[]): void {
+  const [eventName, ...rest] = args
+  eventEmitters.get(channel)?.emit(eventName, ...rest)
+}
+
+function enterState (channel: Channel, nextState: ChannelFsm): void {
+  if (nextState === null || nextState === undefined) {
```

Comment:
this is equivalent to  `if (nextState == null) {` and could be replaced be that.